intake_xarray/image.py

```diff
@@ -0,0 +1,147 @@
+import xarray as xr
+from .base import XarraySource
+
+
+def add_leading_dimension(x):
+    return x[None, ...]
+
+
+def dask_imread(filename, imread=None, preprocess=None):
+    """ Read a stack of images into a dask array
+    Parameters
+    ----------
+    filename: string, iter
+        A globstring like 'myfile.*.png', a list of files, or anything
+        that can go into imread.
+    imread: function (optional)
```

Comment:
Do we support passing functions through? If yes, probably need some careful documentation on how to do that, and how to express it in a catalogue (forgive me, if this is already in the doc/example). If not, can simplify this function.